packages/composites/glv-token/src/transport/base.ts

```diff
@@ -0,0 +1,376 @@
+    result: number,
+    minimizedValue: number,
+    maximizedValue: number,
+    sources: Record<string, string[]>,
+    timestamps: {
+      providerDataRequestedUnixMs: number
+      providerDataReceivedUnixMs: number
+      providerIndicatedTimeUnixMs: undefined
+    },
+  ): AdapterResponse<T['Response']>
+
+  async tokenInfo() {
+    const requestConfig = {
+      url: this.settings.TOKEN_INFO_API,
+      method: 'GET',
+      timeout: this.settings.GLV_INFO_API_TIMEOUT_MS,
+    }
+
+    logger.info('Fetching token info')
+    const response = await this.requester.request<{ tokens: Token[] }>(
+      JSON.stringify(requestConfig),
+      requestConfig,
+    )
+
+    const data: Token[] = response.response.data.tokens
+    data.map((token) => {
+      this.tokensMap[token.address] = token
+      this.decimals[token.symbol] = token.decimals
+    })
+  }
+
+  async marketInfo() {
+    const requestConfig = {
+      url: this.settings.MARKET_INFO_API,
+      method: 'GET',
+      timeout: this.settings.GLV_INFO_API_TIMEOUT_MS,
+    }
+
+    logger.info('Fetching market info')
+    const response = await this.requester.request<{ markets: Market[] }>(
+      JSON.stringify(requestConfig),
+      requestConfig,
+    )
+
+    const data: Market[] = response.response.data.markets
+    data.map((market) => {
+      this.marketsMap[market.marketToken] = market
+    })
+  }
+
+  async _handleRequest(param: RequestParams): Promise<AdapterResponse<T['Response']>> {
+    const providerDataRequestedUnixMs = Date.now()
+    const glv_address = param.glv
+
+    const glvInfo = await this.glvReaderContract.getGlvInfo(
+      this.settings.DATASTORE_CONTRACT_ADDRESS,
+      glv_address,
+    )
+
+    const glv: glvInformation = {
+      glvToken: glvInfo.glv.glvToken,
+      longToken: mapSymbol(glvInfo.glv.longToken, this.tokensMap),
+      shortToken: mapSymbol(glvInfo.glv.shortToken, this.tokensMap),
+      markets: {},
+    }
+
+    for (let i = 0; i < glvInfo.markets.length; i++) {
+      glv.markets[glvInfo.markets[i]] = mapSymbol(glvInfo.markets[i], this.marketsMap)
+    }
+
+    const assets: Array<string> = [glv.longToken.symbol, glv.shortToken.symbol]
+    Object.keys(glv.markets).forEach((m) => {
+      assets.push(mapSymbol(glv.markets[m].indexToken, this.tokensMap).symbol)
+    })
+
+    assets.sort()
+    const priceResult = await this.fetchPrices([...new Set(assets)], providerDataRequestedUnixMs)
+
+    const indexTokensPrices: Array<string | number>[] = []
+    Object.keys(glv.markets).forEach((m) => {
+      const symbol = mapSymbol(glv.markets[m].indexToken, this.tokensMap).symbol
+      indexTokensPrices.push([priceResult.prices[symbol].bid, priceResult.prices[symbol].ask])
+    })
+
+    const glvTokenPriceContractParams = [
+      this.settings.DATASTORE_CONTRACT_ADDRESS,
+      glvInfo.markets,
+      indexTokensPrices,
+      [priceResult.prices[glv.longToken.symbol].bid, priceResult.prices[glv.longToken.symbol].ask],
```

Comment:
Is there are reason why the order was changed ? ask,bid -> bid, ask
Not sure if that affects the logic 
https://github.com/gmx-io/gmx-synthetics/blob/main/contracts/price/Price.sol#L10

Reply:
@karen-stepanyan yes I believe we had it reversed. You pointed out the input is expecting 
```
    struct Props {
        uint256 min;
        uint256 max;
    }
```
This would translate to `bid, ask` in input order to match with `min, max`. 

Here is log capture from the Adapter currently on `main` branch 

```
maximizedValue BigNumber { _hex: '0x0e7b3ac9809d8c1e408b8a892d', _isBigNumber: true }
minimizedValue BigNumber { _hex: '0x0e7bdc9db50951651f3b4142e1', _isBigNumber: true }
maximizedPrice 1.1473320006734626
minimizedPrice 1.1475276398391112
```

Notice here that maximized < minimized

Reply:
I just noticed that the test data in adapter.test.ts was also flipped - so I plugged the above captured values in there

Reply:
thanks @akuzni2